### PR TITLE
Remove dev feature flag for GET /requests API

### LINKED ISSUE #2139: Remove dev feature flag from GET /requests API
### Issue Description

This API has been live on prod for a long time but the feature flag has not been removed.
Remove the dev FF from backend and all other repos where its being used

**Usages in other repos**
1. [website-api-contracts](https://github.com/Real-Dev-Squad/website-api-contracts) - All usages are using dev feature flag :white_check_mark:
2. [website-my](https://github.com/Real-Dev-Squad/website-my) - All usages are using dev feature flag :white_check_mark:
3. [website-dashboard](https://github.com/Real-Dev-Squad/website-dashboard) - All usages are using dev feature flag :white_check_mark: 

This is a subtask of #2075 
### Screenshots

NA
### Reproducibility

 - [ ] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
NA

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [X] Low

### Checklist

- [X] I have read and followed the project's code of conduct.
- [X] I have searched for similar issues before creating this one.
- [X] I have provided all the necessary information to understand and reproduce the issue.
- [X] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by samarpan1738**
Date: 13 Sep 2024

Developer Name: Samarpan Harit

---

## Issue Ticket Number

Closes #2139 
## Description
1. Remove `dev` query param from the next page link being set in the response
2. Make the `dev` query param optional in the request validator
3. Update integration tests

### Documentation Updated?

- [ ] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [X] No

### Database Changes

- [ ] Yes
- [X] No

### Breaking Changes

- [ ] Yes
- [X] No

### Development Tested?

- [X] Yes
- [ ] No

## Screenshots
<details>
<summary>Get requests with `dev` FF</summary>

Screenshot from 2024-09-21 01-47-07
</details>

<details>
<summary>Get requests without `dev` FF</summary>

Screenshot from 2024-09-21 01-47-28
</details>

## Test Coverage
NA

### CODE REVIEW

**Comment by samarpan1738** [comment]
> LGTM, but please verify on local also

I have tested on local with and without `dev` feature flag. Working as expected

File: test/unit/middlewares/requests.test.ts [hunk 1]
```diff
@@ -93,9 +93,7 @@ describe("Create Request Validators", function () {
   describe("Get Request Validator", function () {
     it("Should pass validation for a valid get request", async function () {
       req = {
-        query: {
-          dev: "true",
-        },
+        query: {},
```
Comment (vinit717) [inline test/unit/middlewares/requests.test.ts:96]: Can we remove this

Comment (samarpan1738) [reply test/unit/middlewares/requests.test.ts]: Done

File: test/unit/middlewares/requests.test.ts [hunk 2]
```diff
@@ -92,11 +92,7 @@ describe("Create Request Validators", function () {
 
   describe("Get Request Validator", function () {
     it("Should pass validation for a valid get request", async function () {
-      req = {
-        query: {
-          dev: "true",
-        },
-      };
+      req = {};
```
Comment (vinit717) [inline test/unit/middlewares/requests.test.ts:95]: Here also

**Review by vinit717 — APPROVED** [review]

**Review by fakhruddinkw — APPROVED** [review]

**Pull request merged**